Replace TT1-Blocks with Empty Theme in the wp-env of gutenberg and CI

youknowriad opened the pull request:
At this point, TT1-Blocks is starting to be outdated, we should consider removing it from the Gutenberg dev env and choosing a better theme for development.

We also now have 2022 that is present in the default environment now that it's merged in WordPress trunk.

That said, I decided to replace tt1-blocks with emptytheme here because I think it's the best theme for development, it doesn't have theme styles that can impact behavior of blocks... It's also unopinonated about everything. We still have 2022 in the environment if we want to work with a styled theme but for CI, emptytheme is probably a better default.

WDYT?

I also noticed that in our docs, we still recommend tt1-blocks to test FSE, we should probably update this recommendation separately. 

**testing instructions**

 - Reset your wp-env: `npm run wp-env destroy && npm run wp-env start`
 - Make sure you see the "emptytheme" in the available themes under `Appearance`.

youknowriad committed:
Replace TT1-Blocks with Empty Theme in the wp-env of gutenberg and CI

youknowriad commented on .wp-env.json:
```diff
@@ -1,7 +1,7 @@
 	"core": "WordPress/WordPress",
 	"plugins": [ "." ],
-	"themes": [ "WordPress/theme-experiments/tt1-blocks#tt1-blocks@0.4.7" ],
+	"themes": [ "WordPress/theme-experiments/emptytheme#emptytheme#8f5e564b366db75e6067244ef54b91d06d4f8d8c" ],
```
We could probably use a new tag from that repo.

Mamaduka replied:
Creating a tag sounds like a good idea, probably after porting changes that are needed to the theme.

jasmussen commented:
I like this idea a lot, Empty Theme is an excellent tool to verify whether issues come from theme or editor styles, or whether they come from upstream. So much in fact that it'd make sense to include "Test in Empty Theme" as a suggestion in the PR review instructions.

That being said, I'd love to see some tiny tweaks to the Empty Theme directly to help test some of the theme.json properties, for example "Font Size" is not accessible there:
<img width="1392" alt="Screenshot 2021-12-16 at 10 48 21" src="https://user-images.githubusercontent.com/1204802/146348347-645a9c47-6c0a-4248-8f54-0deef03938b0.png">

gziolo commented:
> So much in fact that it'd make sense to include "Test in Empty Theme" as a suggestion in the PR review instructions.

100% agree - assuming that the Empty Theme keeps its configuration the same forever. In the past we had to patch every theme or tests so they continue passing after modification were applied. This looks like a way to resolve all those issues.

carolinan commented:
Would it make sense to make the test theme part of this repository?
It is now relying on people with access to the theme experiments repo.

youknowriad commented:
> Would it make sense to make the test theme part of this repository?

Definitely something we can consider. What do you all think? @WordPress/gutenberg-core

aristath commented:
+1 from me. Makes sense to have the empty-theme here if we use it for tests etc.

Mamaduka commented:
It makes sense to me as well. We're already keeping plugins required for tests.

A small bonus will be that symlinking is faster than pulling repo from Github.

kjellr commented:
Yeah, I think making it part of the repo here would be totally fine. 👍 

> That being said, I'd love to see some tiny tweaks to the Empty Theme directly to help test some of the theme.json properties, for example "Font Size" is not accessible there:

Oh right, good catch. This PR should fix it: https://github.com/WordPress/theme-experiments/pull/295

ockham commented:
Would definitely love to have whatever theme we use to test FSE (and default to for wp-env) included in this repo. Empty Theme sounds like a good (and lightweight) choice.

ockham commented:
This is currently failing with

```
GitError: fatal: couldn't find remote ref emptytheme#8f5e564b366db75e6067244ef54b91d06d4f8d8c
```

We had a similar error before with TT1 Blocks, since apparently our git library doesn't understand SHAs.

How about we fix this for now by tagging `theme-experiments` with `emptytheme@1.0.0` (to be consistent with our `tt1-blocks` tags)? (Maybe @kjellr or @carolinan have write access to that repo 🙏 ) We can then sort out any potentially remaining issues before copying the theme over into this repo.

kjellr commented:
> How about we fix this for now by tagging theme-experiments with emptytheme@1.0.0 (to be consistent with our tt1-blocks tags)? (Maybe @kjellr or @carolinan have write access to that repo 🙏 ) We can then sort out any potentially remaining issues before copying the theme over into this repo.

Sure thing, here you go!

https://github.com/WordPress/theme-experiments/releases/tag/emptytheme%401.0.0

ockham committed:
Pin emptytheme to v1.0.0

youknowriad committed:
Embed the empty theme in the Gutenberg repository

carolinan commented:
Can you please include testing instructions? :)

youknowriad commented:
@carolinan Sure I added some instructions but it's mostly about the e2e tests passing.

youknowriad commented:
~I'm seeing some issues where the theme sometimes disappear from the available themes. I think it's probably because we load the root folder as a plugin and then a sub folder as a theme which breaks docker volumes somehow. I'll do more tests but maybe we should avoid bundling the theme for now.~

~We could reconsider if we refactor the repo to have two separate folders for the "plugin" and the "theme".~

I'm potentially wrong here, I was switching branches :P

youknowriad committed:
Fix some e2e tests

youknowriad committed:
Fix more e2e tests

youknowriad committed:
Skip some parts of the template revert e2e test

youknowriad commented on packages/e2e-tests/specs/site-editor/template-revert.test.js:
```diff
@@ -164,14 +164,15 @@ describe( 'Template Revert', () => {
 		await revertTemplate();
 		await save();
 		await undoRevertInHeaderToolbar();
+		// there's a bug which probably also exists where the redo button stays disabled.
```
I skipped this test because there's an undo/redo bug that is only visible with the empty theme for some reason but not with the tt1-blocks theme. 

cc @ntsekouras if you have any ideas here.

ntsekouras replied:
These bug are so fun 😢 . I've not found it yet, but I'll keep looking. It's so weird that it behaves differently on themes..

youknowriad replied:
I think it's related to this https://github.com/WordPress/gutenberg/pull/35892#discussion_r780958187

youknowriad committed:
Fix multi entity saving flow test

youknowriad committed:
Skip failing navigation block

youknowriad commented on packages/e2e-tests/specs/editor/blocks/navigation.test.js:
```diff
@@ -435,7 +435,7 @@ describe( 'Navigation', () => {
 	} );
 
-	it( 'allows pages to be created from the navigation block and their links added to menu', async () => {
+	it.skip( 'allows pages to be created from the navigation block and their links added to menu', async () => {
```
@talldan As seen on the latest commits on trunk, this test is failing too often, I'm skipping it for now.

Mamaduka replied:
It's been failing for a while now. I suggested skipping it in [slack last week](https://wordpress.slack.com/archives/C02QB2JS7/p1640246315208600).

ryanwelcher replied:
I've created #37729 that skips this test as it's blocking #36746 as well.

talldan replied:
Apologies, I did get some failures on this when working on it, but I thought I'd solved them all as they passed multiple times locally.

Some discussion on fixing the test - https://github.com/WordPress/gutenberg/pull/37755.

youknowriad commented:
The performance tests here will fail because the "emptytheme" is not on trunk, I think we should probably just ignore that failure and merge.

Mamaduka commented:
> The performance tests here will fail because the "emptytheme" is not on trunk, I think we should probably just ignore that failure and merge.

Does this mean that the "Compare performance with current WordPress Core and previous Gutenberg versions" check will fail as well once we merge this into the trunk?

youknowriad commented:
> Does this mean that the "Compare performance with current WordPress Core and previous Gutenberg versions" check will fail as well once we merge this into the trunk?

Good question, potentially yes, I wonder if we should limit it to the post editor for now in trunk?

youknowriad commented:
Or alternatively, we could decide to use the ".wp-env` file from the "perf tests branch" instead of the one from the plugin's branch. I guess this won't work because the empty theme folder doesn't exist in the old versions.

Mamaduka commented:
Sorry, I don't have a solution right now. I was just thinking out loud 😄

youknowriad committed:
Update the performance job to use the same environment config

youknowriad committed:
Fix issues with the environment

youknowriad committed:
Add missing plugins in the perf test environment

youknowriad committed:
Add some documentation about the performance job

youknowriad commented:
I've reworked the performance job (and added some docs about how it works) but it seems it's still failing from time to time on the site editor thought 🤔 (and not specifically on trunk)

youknowriad committed:
Fix typo

youknowriad commented:
Ok I've managed to make the performance tests (document them and improve how they work). We should be good to go here.

Mamaduka reviewed (approved):
🚢

Mamaduka commented:
It looks like Performance tests are failing on the trunk after this got merged.
